pyci/rdm/algorithms.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np 
+from abc import ABCMeta, abstractmethod
+class Projection(metaclass=ABCMeta):
```

Comment:
We don't need to inherit meta class in here, the regular ABC class should be good. Meta classes gives you ability to overload methods for different types, and I don't think it's a right strategy here.

Just do ABC https://docs.python.org/3/library/abc.html#abc.ABC